### ISSUE TITLE
Segmentation Fault branch 'C' under Linux

### ISSUE THREAD

**Issue by ren2r** [kozec/sc-controller#483]
I'm trying to run the `scc-daemon` on linux but  I got a segmentation fault after the daemon starts and I was unable to figure out how to solve it.
System: `Linux 4.19.59-1-MANJARO`
GLibc: `2.29-3.1`

```bash
[ren2r@psr sc-controller]$ SCC_SHARED=$(pwd) build/src/daemon/scc-daemon debug
[I] Daemon       Starting SC Controller Daemon v0.4.5...
[I] Daemon       Created control socket /home/ren2r/.config/scc/daemon.socket
[I] Daemon       Connected to XServer :0
[I] Daemon       Initializing drivers...
[I] remotepad    Listening on 0.0.0.0:55400
[E] Daemon       Failed to load 'fake'
[D] Mapper       Creating virtual devices...
[d] Mapper       Gamepad:  <UInput 0xf: Microsoft X-Box 360 pad>
[d] Mapper       Mouse:    <UInput 0x10: SC Controller Mouse>
[d] Mapper       keyboard: <UInput 0x11: SC Controller Keyboard>
[I] CemuHook     Created CemuHookUDP Motion Provider
[I] Daemon       Activated default profile '/usr/share/scc/default_profiles/Desktop.sccprofile'
[I] Daemon       Ready.

Caught SIGSEGV, segmentation fault on:
  [0x000000004A7E] le)(+0x4a7e) [0x5574bf8f9a7e]
  [0x29EE2135CEE3] libc.so.6(__libc_start_main+0xf3) [0x7f62e0c51ee3]
  [0x000000006DB0] le)(+0x6db0) [0x5574bf8fbdb0]
  [0x0000000088FB] le)(+0x88fb) [0x5574bf8fd8fb]
  [0x000000009E1C] le)(+0x9e1c) [0x5574bf8fee1c]
  [0x000000004DC2] le)(+0x4dc2) [0x5574bf8f9dc2]
  [0x29EE213C99CE] libc.so.6(+0x939ce) [0x7f62e0cbe9ce]

```

and `dmesg` only output this:
```bash
[ 5454.187709] input: Microsoft X-Box 360 pad as /devices/virtual/input/input35
[ 5454.187949] input: SC Controller Mouse as /devices/virtual/input/input36
[ 5454.188199] input: SC Controller Keyboard as /devices/virtual/input/input37
```

**Comment by kozec** [comment]
Sadly, that backtrace doesn't work as I intended. Can you try running it under gdb?

```
$ SCC_SHARED=$(pwd) gdb build/src/daemon/scc-daemon
> run debug
... and after it crashes
> bt
```

**Comment by ren2r** [comment]
I got this output:
```bash
(gdb) run debug
Starting program: /home/ren2r/sc-controller/build/src/daemon/scc-daemon debug
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[I] Daemon       Starting SC Controller Daemon v0.4.5...
[I] Daemon       Created control socket /home/ren2r/.config/scc/daemon.socket
[New Thread 0x7ffff75fc700 (LWP 7238)]
[I] Daemon       Connected to XServer :0
[I] Daemon       Initializing drivers...
[I] remotepad    Listening on 0.0.0.0:55400
[E] Daemon       Failed to load 'fake'
[D] Mapper       Creating virtual devices...
[d] Mapper       Gamepad:  <UInput 0xf: Microsoft X-Box 360 pad>
[d] Mapper       Mouse:    <UInput 0x10: SC Controller Mouse>
[d] Mapper       keyboard: <UInput 0x11: SC Controller Keyboard>
[I] CemuHook     Created CemuHookUDP Motion Provider
[I] Daemon       Activated default profile '/home/rpaiva/sc-controller/share/default_profiles/Desktop.sccprofile'
[I] Daemon       Ready.

Thread 1 "scc-daemon" received signal SIGSEGV, Segmentation fault.
0x00007ffff785f9ce in __strcmp_sse2_unaligned () from /usr/lib/libc.so.6
(gdb) bt
#0  0x00007ffff785f9ce in __strcmp_sse2_unaligned () from /usr/lib/libc.so.6
#1  0x0000555555558dc2 in on_data_ready (d=0x555555568580 <_daemon>, fd=5, userdata=0x0) at ../src/daemon/device_monitor_linux.c:72
#2  0x000055555555de1c in sccd_poller_mainloop (d=0x555555568580 <_daemon>) at ../src/daemon/poller.c:54
#3  0x000055555555c8fb in sccd_start () at ../src/daemon/daemon.c:562
#4  0x000055555555adb0 in main (argc=1, argv=0x7fffffffddd8) at ../src/daemon/main.c:235

```

**Comment by kozec** [comment]
Yeah, all values from udev may be NULL there. Thanks for reporting :)

**Comment by kozec** [comment]
(fixed in ae16e8766f635e44087e95789f7b1653d46f71bb , I misstyped issue number)
